web/src/components/apps/AppDetailPage.tsx

```diff
@@ -559,25 +559,23 @@ function AppDetailPage(props: Props) {
                   />
                 )}
                 {/* snapshots redirects */}
-                <Redirect
-                  exact
-                  from="/app/:slug/snapshots"
-                  to="/snapshots/partial/:slug"
+                <Route
+                  path="/app/:slug/snapshots"
+                  render={() => <Redirect to="/snapshots/partial/:slug" />}
                 />
-                <Redirect
-                  exact
-                  from="/app/:slug/snapshots/schedule"
-                  to="/snapshots/settings?:slug"
+                <Route
+                  path="/app/:slug/snapshots/schedule"
+                  render={() => <Redirect to="/snapshots/settings?:slug" />}
                 />
-                <Redirect
-                  exact
-                  from="/app/:slug/snapshots/:id"
-                  to="/snapshots/partial/:slug/:id"
+                <Route
+                  path="/app/:slug/snapshots/:id"
+                  render={() => <Redirect to="/snapshots/partial/:slug/:id" />}
                 />
-                <Redirect
-                  exact
-                  from="/app/:slug/snapshots/:id/restore"
-                  to="/snapshots/partial/:slug/:id/restore"
+                <Route
+                  path="/app/:slug/snapshots/:id/restore"
+                  render={() => (
+                    <Redirect to="/snapshots/partial/:slug/:id/restore" />
+                  )}
```

Comment:
Following the insruction laid out here: https://reactrouter.com/en/v6.3.0/upgrading/v5#remove-redirects-inside-switch
<img width="648" alt="Screen Shot 2022-11-16 at 09 26 22" src="https://user-images.githubusercontent.com/4998130/202222029-d06295d4-a42d-497a-823d-55dc633e329c.png">